### DISCUSSION TITLE
Calculate Union on meshes from meshio

### DISCUSSION THREAD

**Discussion by gatoniel** [nschloe/pygalmesh#192] [General]
Hi,
I hope it is okay to ask for help here. Additionally, I am unsure if this is a replicate of #131 

I want to calculate the union of two 3d closed objects defined by triangles. With `meshio.Mesh` it is super straight forward to create the meshes of these objects. But I cannot pass them to `pygalmesh.Union`. Is there a way to use `pygalmesh.Union` on these meshes or how would you proceed? Or is `pygalmesh.Union` only available for primitive objects like Spheres, etc. (i.e., things that can be represented with `_pygalmesh.DomainBase`)?

I would like to use this pseudo code:
```
# verts1, verts2= ...
# faces = ...
mesh1 = meshio.Mesh(
        verts1, [("triangle", faces)]
)
mesh2 = meshio.Mesh(
        verts2, [("triangle", faces)]
)

# do some intermediate stuff to transform mesh1 and mesh2 to correct format, i.e., _pygalmesh.DomainBase

union = pygalmesh.Union([mesh1, mesh2])
```
This code results in this error message:
```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. _pygalmesh.Union(arg0: List[_pygalmesh.DomainBase])

Invoked with: [<meshio mesh object>
  Number of points: 96
  Number of cells:
    triangle: 188, <meshio mesh object>
  Number of points: 96
  Number of cells:
    triangle: 188]
```
If `pygalmesh` is not the correct choice for this, do you know any other package that can do something like this?

**Comment by nschloe** [comment]
pygalmesh's Union only works on domains (the geometry, e.g., sphere centered at X with radius R), not meshes. If you already have your mesh, you're out of pygalmesh's domain.

**Comment by gatoniel** [comment]
Ok, thanks. I thought so already. Do you have any suggestions on how to calculate Unions of predefined meshes? I have three ideas:
1. The discussed ideas in #131 Can you tell me how complicated the PR for this issue is? I can program in Python, but I have never incorporated C functions into python.
2. I convert my predefined meshes somehow back to domains
3. I convert my meshes into voxel images. There I can simply create a union. Then I can simply recreate a mesh based on this voxel representation with https://github.com/nschloe/pygalmesh#meshes-from-numpy-arrays-representing-3d-images

**Comment by gatoniel** [comment]
Hi,
I found another mesh package: https://github.com/mikedh/trimesh
If blender or openSCAD are installed this package can calculate boolean operations on triangular meshes. This is what I need, so we can close this discussion.
Best,
Niklas
